components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/WSDL11SOAPOperationExtractor.java

```diff
@@ -0,0 +1,431 @@
+import javax.wsdl.Operation;
+import javax.wsdl.Output;
+import javax.wsdl.Part;
+import javax.wsdl.Types;
+import javax.wsdl.extensions.schema.Schema;
+import javax.wsdl.extensions.soap.SOAPBinding;
+import javax.wsdl.extensions.soap.SOAPOperation;
+import javax.wsdl.extensions.soap12.SOAP12Binding;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * class that reads wsdl soap operations and maps with the types
+ */
+public class WSDL11SOAPOperationExtractor implements WSDLSOAPOperationExtractor {
+    private static final Logger log = LoggerFactory.getLogger(WSDL11SOAPOperationExtractor.class);
+
+    private String[] primitiveTypes = {"string", "byte", "short", "int", "long", "float","double","boolean"};
+    private List primitiveTypeList = Arrays.asList(primitiveTypes);
+
+    private final String WSDL_ELEMENT_NODE = "element";
+    private static final String WSDL_VERSION_11 = "1.1";
+
+    private Definition wsdlDefinition;
+    private String targetNamespace;
+
+    private List typeList = null;
+    private List<Node> elemList = new ArrayList<>();
+
+    private static volatile APIMWSDLReader wsdlReader;
+
+    public WSDL11SOAPOperationExtractor(APIMWSDLReader wsdlReader) {
+        WSDL11SOAPOperationExtractor.wsdlReader = wsdlReader;
+    }
+
+    @Override
+    public boolean init(byte[] wsdlContent) throws APIMgtWSDLException {
+        boolean canProcess;
+        try {
+            wsdlDefinition = wsdlReader.getWSDLDefinitionFromByteContent(wsdlContent);
+            canProcess = true;
+            targetNamespace = wsdlDefinition.getTargetNamespace();
+            Types types = wsdlDefinition.getTypes();
+            typeList = types.getExtensibilityElements();
+
+            for(Object ext : typeList) {
+                if(ext instanceof Schema) {
+                    Schema schema = (Schema)ext;
+                    Element schemaElement = schema.getElement();
+                    String nodeName = schemaElement.getNodeName();
+                    String nodeNS = nodeName.split(SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_SEPARATOR)[0];
+                    String complexTypeElement = nodeNS + SOAPToRESTConstants.COMPLEX_TYPE_NODE_NAME;
+                    NodeList nodeList = schemaElement.getElementsByTagName(complexTypeElement);
+                    elemList.addAll(SOAPOperationBindingUtils.list(nodeList));
+                }
+            }
+            elemList = reformComplexTypes(elemList);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully initialized an instance of " + this.getClass().getSimpleName()
+                        + " with a single WSDL.");
+            }
+        } catch (APIManagementException e) {
+            log.error("Cannot process the WSDL by " + this.getClass().getName(), e);
+            canProcess = false;
```

Comment:
Can we proceed without WSDL being processed ?

Reply:
It will shown from the front end that wsdl cannot be processed